spynnaker/pyNN/models/spike_source/spike_source_poisson_vertex.py

```diff
@@ -382,10 +558,11 @@ def _write_poisson_parameters(
         self.__n_data_specs += 1
 
         # Write the number of microseconds between sending spikes
-        total_mean_rate = numpy.sum(self.__rate)
+        all_rates = numpy.fromiter(_flatten(self.__data["rates"]), numpy.float)
+        total_mean_rate = numpy.sum(all_rates)
         if total_mean_rate > 0:
             max_spikes = numpy.sum(scipy.stats.poisson.ppf(
-                1.0 - (1.0 / self.__rate), self.__rate))
+                0.999, all_rates))
```

Comment:
Does this work properly for (very) low rate values?  (Also, from experience, 0.999 may lead to problems with accuracies in tests here as it did in the FixedProbabilityConnector...)

(Obviously too this whole file needs to be merged with the recent changes to SSP with a bit of care...)

Reply:
I think this was changed temporarily to get it working.  I have added a master-equivalent now...